### PR TITLE
Add searching for libnuma in CMake

### PR DESCRIPTION
**Pull request by ldorau**

### CODE REVIEW

File: cmake/FindLIBNUMA.cmake [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+message(STATUS "Checking for module 'libnuma' w/o PkgConfig")
+
+find_library(LIBNUMA_LIBRARY NAMES libnuma numa)
+set(LIBNUMA_LIBRARIES ${LIBNUMA_LIBRARY})
```
Comment (bratpiorka) [inline cmake/FindLIBNUMA.cmake:8]: do we need to introduce this new variable? LIBNUMA_LIBRARY isn't enough?

File: cmake/FindLIBNUMA.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+message(STATUS "Checking for module 'libnuma' w/o PkgConfig")
```
Comment (bratpiorka) [inline cmake/FindLIBNUMA.cmake:5]: "w/o PkgConfig" -> "using find_library()" sounds better IMO

File: .github/workflows/pr_push.yml [hunk 3]
```diff
@@ -71,7 +71,7 @@ jobs:
       - name: Install apt packages
         run: |
           sudo apt-get update
-          sudo apt-get install -y cmake clang-format-15
+          sudo apt-get install -y cmake clang-format-15 libnuma-dev
```
Comment (lukaszstolarczuk) [inline .github/workflows/pr_push.yml:74]: I don't think it's required in coding style - perhaps just disable building of the tests

Comment (ldorau) [reply .github/workflows/pr_push.yml]: Done.

File: cmake/FindLIBNUMA.cmake (hunk 1 above)
Comment (ldorau) [reply cmake/FindLIBNUMA.cmake:8]: So what do you suggest? It works as is now.

**Review by lukaszstolarczuk — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**